### PR TITLE
Enhancement/Phone number format validation and parsing

### PR DESCRIPTION
**Pull request by EmCousin**
This PR deals with https://github.com/hostolab/covidliste/issues/326 and https://github.com/hostolab/covidliste/issues/218
- [x] Formats incoming phone numbers to the international e164 format before validation
- [x] Validates the phone number has a mobile format
- [x] Validates French and foreign numbers
- [x] User#human_friendly_phone_number returns a number:
  - National format if French ("06 12 34 56 78")
  - e164 international format if other ("+491625555223")
- [x] Provides a rake task `rake format:phone_numbers` to format all user phone numbers currently present in the database

### Notes
- This PR uses [Phonelib](https://github.com/daddyz/phonelib) (MIT license)
- This PR could be conflicting with https://github.com/hostolab/covidliste/pull/340

### CODE REVIEW

**Comment by carsso** [comment]
That's WAY BETTER than the current implementation, or mine in #340

File: Gemfile [hunk 1]
```diff
@@ -42,6 +42,7 @@ gem "pagy"
 gem "pg_query", ">= 0.9.0"
 gem "pg_search"
 gem "pghero"
+gem "phonelib"
```
Comment (rthbound) [inline Gemfile:45]: ➕

File: app/models/concerns/has_phone_number_concern.rb [hunk 2]
```diff
@@ -4,45 +4,28 @@ module HasPhoneNumberConcern
...
+      allow_blank: false
+    }
   end
 
   def human_friendly_phone_number
-    phone_number.gsub(/^33/, "0").match(/(\d{2})(\d{2})(\d{2})(\d{2})(\d{2})/)&.captures&.join(" ")
+    if parsed_phone_number.valid_for_country?(:fr)
+      parsed_phone_number.national
+    else
+      parsed_phone_number.e164
```
Comment (rthbound) [inline app/models/concerns/has_phone_number_concern.rb:18]: thoughts for always using e164?

File: spec/factories/sequence_factory.rb [hunk 3]
```diff
@@ -9,6 +9,16 @@
   sequence(:lon) { Faker::Address.longitude }
   sequence(:name) { Faker::Name.name }
   sequence(:password) { Faker::Lorem.characters(number: 10) }
-  sequence(:french_phone_number) { Faker::PhoneNumber.phone_number }
+  sequence(:french_phone_number) do
+    # Faker generates cell phones that are not valid
+    # because not yet attributed
+    # (Example : 0729999999 and lower are not attributed, 0730000000 and higher are)
+    # We're looping until Faker generates a valid number
+    phone_number = Faker::PhoneNumber.cell_phone
```
Comment (rthbound) [inline spec/factories/sequence_factory.rb:17]: If faker often gives bad phone numbers, should we use a static phone number?

File: app/models/concerns/has_phone_number_concern.rb (hunk 2 above)
Comment (EmCousin) [reply app/models/concerns/has_phone_number_concern.rb:18]: The phone number is always in e164 in the database with this PR. This instance method is just a convenient way to display the phone number in the view for the user.
Your point's valid though. But it's more a feature debate that's worth discussing in the linked issue in my opinion

File: spec/factories/sequence_factory.rb (hunk 3 above)
Comment (EmCousin) [reply spec/factories/sequence_factory.rb:17]: I considered that, but ultimately I don't have a good feeling about generating the same phone number for every user in the test database. Sure, we're not validating the uniqueness of the phone number per user (yet), but I've determined that Faker generates an invalid _cellphone_ number only 1 time out of three, which makes this loop not so much of a performance problem.

File: app/models/concerns/has_phone_number_concern.rb (hunk 2 above)
Comment (mathieuripert) [reply app/models/concerns/has_phone_number_concern.rb:18]: makes sense to display the national version of it for french numbers.

**Review by mathieuripert — APPROVED** [review]
Nice :) Thanks @EmCousin

**Pull request merged**